src/domain/schemas/comment-schemas.js

```diff
@@ -0,0 +1,28 @@
+import joi from 'joi';
+import { idOrKeySchema } from './shared-schemas.js';
+import { maxCommentTextLength, maxCommentTopicLength } from '../validation-constants.js';
+
+export const putCommentBodySchema = joi.object({
+  documentId: idOrKeySchema.required(),
+  topic: joi.string().required(),
+  text: joi.string().required()
+});
+
+export const postCommentBodySchema = joi.object({
+  topic: joi.string().required()
+});
+
+export const deleteCommentBodySchema = joi.object({
+  commentId: idOrKeySchema.required()
+});
+
+export const commentDBSchema = joi.object({
+  _id: idOrKeySchema.required(),
+  documentId: idOrKeySchema.required(),
+  createdOn: joi.date().required(),
+  createdBy: idOrKeySchema.required(),
+  deletedOn: joi.date().allow(null),
+  deletedBy: idOrKeySchema.allow(null),
+  topic: joi.string().max(maxCommentTopicLength).required(),
+  text: joi.string().max(maxCommentTextLength).allow(null)
```

Comment:
I think this is another one of the examples where you like `null` and I don't (as the user can hardly ever enter it), but we can postpone that for an eventual streamlining session.

Reply:
I agree with the empty string default, changed!